app/models/guest.rb

```diff
@@ -4,6 +4,8 @@ class Guest < ApplicationRecord # :nodoc:
   # Associations
   has_many :reservations, dependent: :destroy
 
+  accepts_nested_attributes_for :reservations, allow_destroy: true
```

Comment:
@waywho on this wayI will be able to create a guest within a create reservation form correct?
I will introduce the respective params on the controllers in a separate pr.

Reply:
I think here you do not need the `allow_destroy` option, unless you are already using it within the project.

Reply:
hmmm. This is an interesting one. So when you set `accepts_nested_attributes_for` within the Guest model, the order of things to create in the backend is: First Guest, then Reservation.  So that is how I usually think about it. So technically and pedantically, it is not "creating a guest within a create reservation form". 

But that doesn't mean that this is wrong. This is probably for all intends and purposes correct. Just that your frontend form will need some tweaking when it comes to it. You will be technically needing a guest first in order to create the reservation, instead of creating a reservation first then create a guest, if that makes sense. Of course, we can still make the form looks like you are creating a reservation and creating a guest within the reservation form. And in the controller, we just need to build the guest first then build the reservation.

You can put `accepts_nested_attributes_for` in the Reservation model, but we have to do the same thing in the controller backend when it comes to create because it is then a `belongs_to` association: Build the Guest then the Reservation, because of the Parent(Guest) - Child(Reservation) association.

So where to put it comes down to how you mentally thinking about it. If you are thinking you are creating Reservation first, then creating Guest. i.e. your entry is `ReservationController`, then it might make sense to put this `accepts_nested_attributes_for` in the Reservation Model. This is more of a preference rather than rule, I think. It comes down to if you come back to review this again in 6 months, will it be easy for you to understand how it is all connected up.

Last note `allow_destroy`, allows your to destroy the nested association. It is defaulted to allow you to just edit/update normally. Setting it here means that it allows you to destroy Guest if you want to, and I think what Alberto means is that if you have existing Guest attached to a reservation, you could destroy the Guest within the Reservation form. Is that what you want? Perhaps not?

Reply:
About the allow destroy, let's remove for now, let's focus on the order that things are being created here.
What I have in mind is:
- When the `User` is filling the form to create the reservation, a `Guest` will be required, it should associate an existing guest or create a new one and associate it's id, that's why I'm pretty sure that the guest may be associated/created first. So @waywho, we're talking about the both scenarios you described. "First guest, then reservation. So that is how I usually think about it. So technically and pedantically, it is not "creating a guest within a create reservation form"."

Reply:
I'll create a new issue and let's refine it better there.